kasa/smartdimmer.py

```diff
@@ -41,19 +41,37 @@ def brightness(self) -> int:
         return int(sys_info["brightness"])
 
     @requires_update
-    async def set_brightness(self, value: int):
-        """Set the new dimmer brightness level in percentage."""
+    async def set_brightness(self, brightness: int, transition: int = None):
+        """Set the new dimmer brightness level in percentage.
+
+        :param int transition: duration in milliseconds to transition to the new brightness (default: None)
+        """
         if not self.is_dimmable:
             raise SmartDeviceException("Device is not dimmable.")
 
-        if not isinstance(value, int):
-            raise ValueError("Brightness must be integer, " "not of %s.", type(value))
-        elif 0 <= value <= 100:
+        if not isinstance(brightness, int):
+            raise ValueError(
+                "Brightness must be integer, " "not of %s.", type(brightness)
+            )
+        elif not 0 <= brightness <= 100:
+            raise ValueError("Brightness value %s is not valid." % brightness)
+        elif transition is not None and not isinstance(transition, int):
+            raise ValueError(
+                "Transition must be integer, " "not of %s.", type(transition)
+            )
+        elif transition is not None and transition <= 0:
+            raise ValueError("Transition value %s is not valid." % transition)
```

Comment:
These two can be combined under `if transition is not None:` and checked under that only for the values.